docs/spec/BeakerSpec.tex

```diff
@@ -0,0 +1,1055 @@
+  executing a Register Procedure syscall will be called.
+  \item This code will execute and return whatever value it deems appropriate.
+\end{enumerate}
+
+\subsection{Return and Error Codes}\label{errors}
+This section defines return and error codes for interacting with the kernel.
+These values are returned by system calls \emph{only} in the event of a failure
+of the kernel, in which case the DELEGATECALL will have been reverted. As a
+result, these error codes are provided only when the kernel reverts, if the
+DELEGATECALL returns normally, whatever is returned is determined by the
+procedure being called. That is, the DELEGATECALL should leave a 0 on the stack.
+
+The error code is always 8-bits, and may followed by additional data.
+
+Therefore:
+
+\begin{itemize}
+  \item If 1 is left on the stack after the DELEGATECALL of a system call, then
+  the system call succeeded, and whatever data was returned was returned by
+  system call.
+  \item If 0 is left on the stack after the DELEGATECALL of a system call, then
+  the system call failed, and whatever data was returned is defined in the table
+  below.
+\end{itemize}
+
+\begin{table}[H]
+  \caption{Return and error codes.}
+  \centering{}%
+  \begin{tabularx}{\textwidth}{l|c|X|X}
+    \hline
+    Type & Value & Additional Data & Description \\
+    \hline
+    \hline
+    SYSCALL\_BADCAP  & 33 & None & Capability insufficient. \\
+    SYSCALL\_NOGAS   & 44 & None & Procedure execution ran out of gas. \\
+    SYSCALL\_REVERT  & 55 & Returned data from procedure & The called
+      procedure reverted. \\
+    SYSCALL\_FAIL    & 66 & System call specific error data & The system call
+      failed for reasons specific to the system call. \\
+    SYSCALL\_NOEXIST & 111 & None & The syscall integer values is not a valid
+      syscall as defined in Table \ref{syscall-numbers}. \\
+    \hline
+  \end{tabularx}
+\end{table}
+
+\subsection{Capability Subsets}\label{cap-subsets}
+
+Capability $B$ is a subset\footnote{$A$ subset, not a \emph{strict} or
+\emph{proper} subset.} of Capability $A$, if everything that can be done using
+Capability $B$ can be done using Capabilty $A$. The following rules apply:
+
+\begin{itemize}
+  \item $A$ is always a subset of itself. $A \subseteq A$
+  \item Transitivity: $A \subseteq B \land B \subseteq C \implies A
+  \subseteq C$
+\end{itemize}
+
+As the nature of each capability is different, the way subsets are defined is
+different for each one. The following sections will define how subsets are
+defined for each capability.
+
+\subsection{Call System Call}
+
+\subsubsection{System Call Format}
+Type Value: 3
+
+Data: Note that here the offset is from after system call type.
+
+Note that here we could pack the data much more densely, but we don't in order
+to err on the side of simplicity and stick as close as we can to 32-byte values.
+
+\begin{table}[H]
+  \caption{Call procedure system call format.}
```

Comment:
The `proc#call` system call format is missing the memory slice (64 bytes) payload for the called procedure.

Reply:
Noted I left that off the end, I've updated it to be consistent with what we previously implemented.